src/Operations.jl

```diff
@@ -242,13 +242,10 @@ function collect_project(pkg::PackageSpec, path::String)
         pkgerror("could not find project file for package $(err_rep(pkg)) at `$path`")
     end
     project = read_package(project_file)
-    #=
-    # TODO, this should either error or be quiet
     julia_compat = get_compat(project, "julia")
-    if julia_compat !== nothing && !(VERSION in julia_compat)
-        println(io, "julia version requirement for package $(err_rep(pkg)) not satisfied")
+    if !Base.get_bool_env("JULIA_PKG_JULIA_COMPAT_IGNORE", false) && !isnothing(julia_compat) && !(VERSION in julia_compat)
+        pkgerror("julia version requirement for package $(err_rep(pkg)) not satisfied; you can override and ignore this check by setting `ENV[\"JULIA_PKG_JULIA_COMPAT_IGNORE\"]`")
     end
```

Comment:
```suggestion
    if julia_compat !== nothing && !(VERSION in julia_compat)
        # If the environment variable is not set, or if it is set to "error", we throw an error.
        # If the environment variable is set to "warn", we print a warning, and we continue.
        # If the environment variable is set to "ignore", we do not print any warning, and we continue.
        #
        # If the environment variable is set to any other value, we print a message saying that the
        # user has set an invalid value for the environment variable, and then we proceed as if
        # the environment variable was set to "error", i.e. we throw an error.
        env_var_name = "JULIA_PKG_CHECK_JULIA_COMPAT"
        env_var_value = strip(get(ENV, env_var_name, "error"))
        msg = "Julia version requirement for package $(err_rep(pkg)) not satisfied; you can override and ignore this check by setting `ENV[\"$(env_var_name)\"]`"
        if env_var_value == "warn"
            @warn msg
        elseif env_var_value == "ignore"
        else
            if env_var_value != "error"
                @warn "Invalid value for $(env_var_name): $(env_var_value)"
            end
            pkgerror(msg)
        end
    end
```

Reply:
let me know what is decided on this and I will be happy to do the finishing touches